tests/app/test_modbus_client.py

```diff
@@ -1,134 +1,58 @@
 """Unit tests for the ModbusClient class in the app.modbus_client module."""
 
-import socket
-import threading
-from multiprocessing import Pipe
-
+from unittest.mock import MagicMock
 from pymodbus.client import ModbusTcpClient
 from app.memory_order import MemoryOrder
-
 from app.modbus_client import ModbusClient
-from app.configuration import (
-    Coil,
-    Configuration,
-    HoldingRegister,
-    ModbusSettings,
-    MqttSettings,
-)
+from app.configuration import Coil, Configuration, ModbusSettings, HoldingRegister
+import pytest
 
 
 class TestModbusClient:
     def setup_class(self):
-        # This method will be called before every test
         self.coils = [Coil("test_coil", [1])]
         self.holding_registers = [
-            HoldingRegister("test_register", MemoryOrder("AB"), "INT16", 1.0, [1]),
+            HoldingRegister("int_register", MemoryOrder("AB"), "INT16", 1.0, [1]),
```

Comment:
I wonder if we could increase the coverage here. To cover all the holding registers data types

Reply:
Also, if this is only used by the `test_registers` function, I wonder if we can move the variable down to be part of the test, rather than a class property. Not strongly opinionated here, just wondering what are your thoughts here.

Reply:
The `holding_registers` var is passed to the Configuration object in setup, as well as used in `test_registers`, so needs to be initialised in the class setup.

We could cover more data types here but I'm not actually doing anything different for each one other than sending a different sample value - not checking how the payload is constructed or what args are passed to the mock -  so there probably isn't much value in even sending two. It's easy to assert on the arguments passed for coils, but because int/float values are being converted to a payload, I figured it would be a bit redundant to assert specifically here on how that is happening - the payload_builder tests are already doing that.

Reply:
Thank you for that clarification :)
LGTM